# Exploratory Data Analysis

Este notebook tem como objetivo realizar a análise exploratória dos dados tratados do projeto.

Nesta etapa, serão observados padrões gerais da base, como quantidade de jogos, distribuição de vitórias em casa, médias de pontos e comportamento inicial das equipes jogando em casa e fora de casa.

In [1]:
import pandas as pd

In [2]:
games = pd.read_csv('../data/processed/clean_games.csv')
teams = pd.read_csv('../data/processed/clean_teams.csv')

## Visualização inicial dos dados

Nesta etapa, será feita uma visualização inicial dos datasets tratados para confirmar a estrutura dos dados e revisar as colunas disponíveis.

In [3]:
games.head()

,GAME_DATE_EST,HOME_TEAM_ID,VISITOR_TEAM_ID,SEASON,PTS_home,PTS_away,HOME_TEAM_WINS
0,2022-12-22,1610612740,1610612759,2022,126.0,117.0,1
1,2022-12-22,1610612762,1610612764,2022,120.0,112.0,1
2,2022-12-21,1610612739,1610612749,2022,114.0,106.0,1
3,2022-12-21,1610612755,1610612765,2022,113.0,93.0,1
4,2022-12-21,1610612737,1610612741,2022,108.0,110.0,0


In [4]:
teams.head()

,TEAM_ID,ABBREVIATION,NICKNAME,CITY
0,1610612737,ATL,Hawks,Atlanta
1,1610612738,BOS,Celtics,Boston
2,1610612740,NOP,Pelicans,New Orleans
3,1610612741,CHI,Bulls,Chicago
4,1610612742,DAL,Mavericks,Dallas


## Informações gerais dos datasets

Aqui serão verificadas a quantidade de registros, os tipos de dados e as colunas presentes nos arquivos tratados.

In [5]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26523 entries, 0 to 26522
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   GAME_DATE_EST    26523 non-null  object 
 1   HOME_TEAM_ID     26523 non-null  int64  
 2   VISITOR_TEAM_ID  26523 non-null  int64  
 3   SEASON           26523 non-null  int64  
 4   PTS_home         26523 non-null  float64
 5   PTS_away         26523 non-null  float64
 6   HOME_TEAM_WINS   26523 non-null  int64  
dtypes: float64(2), int64(4), object(1)
memory usage: 1.4+ MB


In [6]:
teams.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   TEAM_ID       30 non-null     int64 
 1   ABBREVIATION  30 non-null     object
 2   NICKNAME      30 non-null     object
 3   CITY          30 non-null     object
dtypes: int64(1), object(3)
memory usage: 1.1+ KB


## Quantidade total de jogos e equipes

Nesta parte, serão identificados o total de partidas presentes na base e a quantidade de equipes disponíveis no dataset.

In [7]:
print('Quantidade total de jogos:', games.shape[0])
print('Quantidade total de equipes:', teams.shape[0])

Quantidade total de jogos: 26523
Quantidade total de equipes: 30


## Distribuição de vitórias do time da casa

A coluna HOME_TEAM_WINS indica se a equipe mandante venceu a partida.
Nesta etapa, será observada a quantidade de vitórias e derrotas dos times jogando em casa.

In [8]:
games['HOME_TEAM_WINS'].value_counts()

HOME_TEAM_WINS
1    15628
0    10895
Name: count, dtype: int64

In [9]:
home_wins = games['HOME_TEAM_WINS'].sum()
home_losses = games.shape[0] - home_wins

print('Vitórias do time da casa:', home_wins)
print('Derrotas do time da casa:', home_losses)

Vitórias do time da casa: 15628
Derrotas do time da casa: 10895


## Médias de pontuação

Aqui serão analisadas as médias de pontos marcados pelas equipes mandantes e visitantes, permitindo uma comparação inicial entre desempenho em casa e fora.

In [10]:
print(f"Média de pontos do time da casa: {games['PTS_home'].mean():.2f}")
print(f"Média de pontos do time visitante: {games['PTS_away'].mean():.2f}")

Média de pontos do time da casa: 103.45
Média de pontos do time visitante: 100.63


## Medidas de Centralização — Moda

Além da média, a **moda** é uma medida de centralização importante: ela indica o valor que aparece com maior frequência na distribuição. Para variáveis contínuas como a pontuação, a moda revela qual placar é mais recorrente nas partidas.

In [11]:
from scipy import stats

moda_home = games['PTS_home'].mode()[0]
moda_away = games['PTS_away'].mode()[0]
mediana_home = games['PTS_home'].median()
mediana_away = games['PTS_away'].median()

print('=== Medidas de Centralização — Pontuação ===')
print(f'Média   do time da casa:   {games["PTS_home"].mean():.2f}')
print(f'Mediana do time da casa:   {mediana_home:.2f}')
print(f'Moda    do time da casa:   {moda_home}')
print()
print(f'Média   do time visitante: {games["PTS_away"].mean():.2f}')
print(f'Mediana do time visitante: {mediana_away:.2f}')
print(f'Moda    do time visitante: {moda_away}')

=== Medidas de Centralização — Pontuação ===
Média   do time da casa:   103.45
Mediana do time da casa:   103.00
Moda    do time da casa:   104.0

Média   do time visitante: 100.63
Mediana do time visitante: 100.00
Moda    do time visitante: 98.0


A **moda** dos pontos dos times da casa e visitantes revela o placar mais comum nas partidas. Quando a moda se aproxima da mediana e da média, a distribuição é aproximadamente simétrica. Diferenças entre esses três valores indicam assimetria na distribuição de pontuações.

## Estatísticas descritivas das pontuações

Nesta parte, serão observadas medidas descritivas das pontuações dos times da casa e visitantes, como média, desvio padrão, mínimo e máximo.

In [12]:
games[['PTS_home', 'PTS_away']].describe().round(2)

,PTS_home,PTS_away
count,26523.00,26523.00
mean,103.45,100.63
std,13.28,13.43
min,36.00,33.00
25%,94.00,91.00
50%,103.00,100.00
75%,112.00,110.00
max,168.00,168.00


## Estatísticas descritivas das pontuações

A tabela acima apresenta estatísticas descritivas das colunas `PTS_home` e `PTS_away`, que representam, respectivamente, os pontos marcados pelo time da casa e pelo time visitante.

A linha **count** indica a quantidade de registros válidos disponíveis em cada coluna, mostrando que a base possui 26.523 jogos com pontuação registrada para ambos os casos.

A linha **mean** representa a média de pontos. Observa-se que os times da casa tiveram média de **103,45 pontos**, enquanto os visitantes tiveram média de **100,63 pontos**, sugerindo um desempenho ofensivo ligeiramente superior das equipes mandantes.

A linha **std** corresponde ao desvio padrão, que mostra o quanto os valores costumam variar em relação à média. Nesse caso, tanto os pontos dos times da casa quanto dos visitantes apresentam variação semelhante, em torno de 13 pontos.

A linha **min** mostra a menor pontuação registrada, enquanto a linha **max** indica a maior pontuação observada em cada coluna.

As linhas **25%**, **50%** e **75%** representam os quartis da distribuição. O valor de **50%** corresponde à mediana, isto é, o ponto central dos dados. Nota-se que a mediana de `PTS_home` é **103**, enquanto a de `PTS_away` é **100**, reforçando a tendência de maior pontuação das equipes jogando em casa.

De modo geral, essa tabela indica que os times mandantes apresentam, em média, uma pontuação um pouco superior à dos visitantes, o que pode ser um primeiro indício de influência do fator casa no desempenho das equipes.

## Medidas de Dispersão — Variância, Desvio Padrão e Amplitude

As medidas de dispersão quantificam o grau de espalhamento dos dados em torno da média. Além do desvio padrão (já presente no `.describe()`), a **variância** e a **amplitude** completam o conjunto mínimo exigido de medidas de dispersão.

In [13]:
print('=== Medidas de Dispersão — Pontuação ===')
for col, label in [('PTS_home', 'Time da Casa'), ('PTS_away', 'Time Visitante')]:
    variancia  = games[col].var()
    desvio_pad = games[col].std()
    amplitude  = games[col].max() - games[col].min()
    print(f'\n{label}:')
    print(f'  Variância:      {variancia:.2f}')
    print(f'  Desvio Padrão:  {desvio_pad:.2f}')
    print(f'  Amplitude:      {amplitude} (min={games[col].min()}, max={games[col].max()})')

=== Medidas de Dispersão — Pontuação ===

Time da Casa:
  Variância:      176.38
  Desvio Padrão:  13.28
  Amplitude:      132.0 (min=36.0, max=168.0)

Time Visitante:
  Variância:      180.42
  Desvio Padrão:  13.43
  Amplitude:      135.0 (min=33.0, max=168.0)


A **variância** mede a dispersão quadrática dos valores em relação à média — quanto maior, mais espalhados estão os pontos. O **desvio padrão** é a raiz quadrada da variância e tem a mesma unidade da variável original (pontos), sendo mais intuitivo para interpretação. A **amplitude** indica a diferença entre o maior e o menor placar registrado na base.

## Distribuição por temporada

A análise por temporada ajuda a entender quantas partidas existem em cada período da base de dados.

In [14]:
games['SEASON'].value_counts().sort_index()

SEASON
2003    1286
2004    1362
2005    1432
2006    1419
2007    1411
2008    1425
2009    1424
2010    1422
2011    1104
2012    1420
2013    1427
2014    1418
2015    1416
2016    1405
2017    1382
2018    1378
2019    1241
2020    1220
2021    1389
2022     542
Name: count, dtype: int64

## Preparação para análises futuras

Nesta etapa, será criada a coluna AWAY_TEAM_WINS, que representa as vitórias da equipe visitante.
Ela será útil nas próximas análises comparativas entre desempenho em casa e fora.

In [15]:
games['AWAY_TEAM_WINS'] = 1 - games['HOME_TEAM_WINS']
games.head()

,GAME_DATE_EST,HOME_TEAM_ID,VISITOR_TEAM_ID,SEASON,PTS_home,PTS_away,HOME_TEAM_WINS,AWAY_TEAM_WINS
0,2022-12-22,1610612740,1610612759,2022,126.0,117.0,1,0
1,2022-12-22,1610612762,1610612764,2022,120.0,112.0,1,0
2,2022-12-21,1610612739,1610612749,2022,114.0,106.0,1,0
3,2022-12-21,1610612755,1610612765,2022,113.0,93.0,1,0
4,2022-12-21,1610612737,1610612741,2022,108.0,110.0,0,1


## Conclusões iniciais da análise exploratória

A análise exploratória permite observar o comportamento geral da base e identificar padrões iniciais relacionados ao fator casa.

A partir desta etapa, já é possível verificar:
- quantidade total de partidas analisadas;
- frequência de vitórias dos times mandantes;
- médias de pontos em casa e fora;
- distribuição dos jogos por temporada.

Essas informações servirão de base para a próxima etapa, em que será feita a análise específica do desempenho das equipes jogando em casa versus fora de casa.